gems/aws-sdk-core/spec/aws/assume_role_credentials_spec.rb

```diff
@@ -174,5 +174,130 @@ module Aws
       expect(before_refresh_called).to be(true)
     end
 
+    it 'defaults the source to none' do
+      c = AssumeRoleCredentials.new(
+        role_arn: 'arn',
+        role_session_name: 'session'
+      )
```

Comment:
could create a let variable that just does `AssumeRoleCredentials` to avoid repeated code within each `it` blocks.

when you need more configurations for different test cases,  you could create a another let variable that uses "adds" more stuff to the previous - something like?

```
let(:creds) do
   AssumeRoleCredentials.new(
        role_arn: 'arn',
        role_session_name: 'session'
   )
end

let(:resolved_creds) do
  c = creds
  c.resolving = true 
  c
end

```